daemon/src/main/java/org/lsposed/lspd/cli/Main.java

```diff
@@ -0,0 +1,151 @@
+package org.lsposed.lspd.cli;
+
+import android.os.Binder;
+import android.os.IBinder;
+import android.os.Parcel;
+import android.os.RemoteException;
+import android.os.ServiceManager;
+
+import com.beust.jcommander.JCommander;
+import com.beust.jcommander.ParameterException;
+
+import org.lsposed.lspd.ILSPManagerService;
+import org.lsposed.lspd.cli.exception.FormatException;
+import org.lsposed.lspd.cli.exception.NotXposedModuleException;
+import org.lsposed.lspd.cli.exception.RunException;
+import org.lsposed.lspd.cli.exception.UninstalledPackageException;
+import org.lsposed.lspd.cli.exception.UnknownCommandException;
+import org.lsposed.lspd.cli.handler.ICommand;
+import org.lsposed.lspd.cli.handler.LogHandler;
+import org.lsposed.lspd.cli.handler.ModulesHandler;
+import org.lsposed.lspd.cli.handler.ScopeHandler;
+import org.lsposed.lspd.cli.handler.StatusHandler;
+import org.lsposed.lspd.service.ILSPApplicationService;
+import org.lsposed.lspd.util.SignInfo;
+
+import static org.lsposed.lspd.cli.Utils.CMDNAME;
+import static org.lsposed.lspd.cli.Utils.ERRCODES;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+
+public class Main {
+
+    private static ILSPManagerService getLSPManagerService() throws RemoteException {
+        var activity_service = ServiceManager.getService("activity");
+        var heartBeat = new Binder();
+        Parcel data = Parcel.obtain();
+        Parcel reply = Parcel.obtain();
+        data.writeInterfaceToken("LSPosed");
+        data.writeInt(2);
+        data.writeString("lsp-cli:" + SignInfo.CLI_UUID);
+        data.writeStrongBinder(heartBeat);
+
+        var resArr = new ArrayList<IBinder>(1);
+        if(activity_service.transact(1598837584, data, reply, 0)) {
+            reply.readException();
+            var serviceBinder = reply.readStrongBinder();
+            if (serviceBinder == null) {
+                System.err.println("binder null");
+                return null;
+            }
+            var service = ILSPApplicationService.Stub.asInterface(serviceBinder);
+            if(service.requestInjectedManagerBinder(resArr) == null) {
```

Comment:
Maybe a specific method like `requestCLIBinder` is better.

Reply:
ok done